### ISSUE TITLE
"Associated reference no longer appears to be available." when using ReferenceInput 

### ISSUE THREAD

**Issue by macrozone** [marmelab/react-admin#3901]
**What you were expecting:**

I have this component that used to work in react-admin 2:

```
export const PageCreate = (props) => (
  <Create title="Create a Page" {...props}>
    <SimpleForm>
      <TextInput disabled source="id" />
      <TextInput source="title" />
      <ReferenceInput
        label="Parent Page"
        source="parentPage.id"
        reference="Page"
        allowEmpty
        filterToQuery={(searchText) => ({ path_contains: searchText })}
      >
        <SelectInput optionText="path" emptyValue={null} />
      </ReferenceInput>
    </SimpleForm>
  </Create>
);
````

I can select a parent page id with the reference input

**What happened instead:**

As soon as i blur the select field, the field is marked as invalid with the following error:

`Associated reference no longer appears to be available.`

**Steps to reproduce:**

I know i should provide a code-sandbox and will do that (this takes some time), but i wanted to ask first if you guys have an idea what the regression might be here. The same code worked in version 2 and i don't see a breaking change that might lead to this error. 

I also tried to debug and i see that the field holds the correct id, which is used as reference to the page. 

I know that others have reported that this error occurse if you don't use a id as a reference (e.g. here https://github.com/marmelab/react-admin/issues/3764), but that's the case here. i clearly use an id.

I also see that the data-provider fires a request (react-admin-open-crud) with the selected id and retrievs the the correct page (in an array).

**Related code:**

will add

```
insert short code snippets here
```

**Other information:**

**Environment**

* React-admin version: 3.0.0-beta.3
* Last version that did not exhibit the issue (if applicable): 2.9.8
* React version: 16.10
* Browser: chrome
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
There is some work related to this #3870

**Comment by macrozone** [comment]
@heesienooi I think it might be better to modifiy `ReferenceInput` so that 

```
 allowNull
 format={(v) => (!v ? null : v)} 
```

are always set as default. A ReferenceInput should always encode "no value selected" as null, not as ""

**Comment by fzaninotto** [comment]
Should be fixed by #3983. Please report if the issue is still there after the next release.
